### ISSUE TITLE
[Shop API] Separate Checkout API from the state machine states

### ISSUE THREAD

**Issue by dbalabka** [Sylius/Sylius#11842]
# TL;DR;
I would like to propose to separate Checkout API from the state machine states:
https://docs.sylius.com/en/1.7/api/checkouts.html
I found that it might be useful to have a single endpoint to create an order or/and fill order partially.

# Problem
The motivation to create such endpoints is to separate View and Backend logic. Use cases of for such separation might be useful if we would like to AB test of possible flows of the checkout process (e.g., single-step checkout versus multiple steps checkout). Unfortunately, in the current implementation, it is not possible due to our backend logic strictly describe checkout steps by the state machine configured states. 

# Solution
In RESTful terms there should be two endpoints:
1. with POST method to create order
2. and PUT/PATCH method to modify the order.

Order finalization might be done during payment creation for a particular order. For example, the order will be marked as paid if payment has been created.

**Comment by kayue** [comment]
What is your suggestion on retrieving eligible shipping method / payment method when we only use 1 API to complete checkout?

By this way your idea is what I implemented for my store.

**Comment by dbalabka** [comment]
@kayue good catch. There  should be another endpoint which returns a list of eligible shipping/payment methods based on partially filled order. If there not enough data then error (e.g., 400) should be returned.
Probably, there might be two options: when we provide all required information with request or take information from order otherwise.

**Comment by dbalabka** [comment]
Do not stale!
